bokeh/palettes.py

```diff
@@ -1563,6 +1567,54 @@ def diverging_palette(palette1: Palette, palette2: Palette, n: int, midpoint: fl
     # return piecewise linear interpolation of colors
     return linear_palette(palette1, n1) + linear_palette(palette2, n2)
 
+def single_color_palette(color: str, n: int | None = None, start_alpha: int = 0, end_alpha: int = 255) -> Palette:
+    """ Generate a palette that is a single color with linearly varying alpha.
+
+    Alpha may vary from low to high or high to low, depending on the values of
+    ``start_alpha`` and ``end_alpha``.
+
+    Args:
+        color (str) :
+            Named color or RGB(A) hex color string. Any alpha component is
+            ignored.
```

Comment:
Looks good. One alternative behavior to consider is whether to use the color's internal alpha value as the default for `end_alpha`, rather than the current behavior of ignoring the color's alpha and then using a fixed default `end_alpha` of 255. My reasoning is that people will normally expect the result of this function to be a palette ranging from 0 opacity up to the specified color, and such an expectation will be violated if (possibly unbeknownst to them) that color has an internal alpha value. Outside this function, colors with an internal alpha value will appear dimmer than those without, and if we throw that alpha value away, we'll be using the full alpha range for that hue but not respecting the idea that it's a range between "none of this color" and "fully this color" as is otherwise the case for this function. On the other hand, discarding this color might be preferred so that the docstring can more clearly describe that existing alpha is simply discarded, which probably makes the mechanism easier to understand (even if surprising in some cases where colors do have alpha). I'll leave that decision to @ianthomas23, but thought it was worth considering, at least.

Reply:
Just to note, the exact behavior of RGBA + A situation across bokehjs (so far) is defined in `core/util/color.ts`. This doesn't necessarily have to influence the behavior in this PR.

Reply:
Thanks for the pointer.

Reply:
I agree that multiplying them would be much more natural and consistent, but I also agree that having the behavior (whatever it is) consistent across Bokeh is more important. So that's a decision I'd leave to @bryevdv and @mattpap , i.e., should Bokeh in general respect the A value of a color in all cases, using that to modulate any provided alpha (or vice versa, depending on how you think about it)? That's the behavior that I'd prefer as being more natural and more useful ("why is the alpha being ignored!!!!!????!?!?!"), but it's a behavior change and thus would need top-level approval.